lumicks/pylake/calibration.py

```diff
@@ -79,18 +81,39 @@ def power_spectrum_params(self):
             "fit_range": self.fit_range,
         }
 
+    def __getitem__(self, item):
+        try:
+            return super().__getitem__(item)
+        except KeyError:
+            return getattr(self, item)
+
     @_verify_full
     def _model_params(self):
         """Returns parameters with which to create an active or passive calibration model"""
 
         # TODO: Model needs to support fixed_diode and fixed_alpha, keep API private until finalized
-        return _read_from_bl_dict(self.data) | {
+        params = {
+            "bead_diameter": self.bead_diameter,
+            "rho_bead": self.rho_bead,
+            "rho_sample": self.rho_sample,
+            "viscosity": self.viscosity,
+            "temperature": self.temperature,
+            "driving_frequency_guess": self.driving_frequency_guess,
+            "distance_to_surface": self.distance_to_surface,
+            # Fixed diode parameters (only available when diode was fixed)
+            "fixed_alpha": self.diode_relaxation_factor if not self.fitted_diode else None,
+            "fixed_diode": self.diode_frequency if not self.fitted_diode else None,
+            # Only available for axial with active
+            "drag": self.transferred_lateral_drag_coefficient,
+            # Other properties
             "sample_rate": self.sample_rate,
             "fast_sensor": self.fast_sensor,
             "axial": bool(self.data.get("Axial calibration")),
-            "hydrodynamically_correct": bool(self.data.get("Hydrodynamic correction enabled")),
+            "hydrodynamically_correct": self.hydrodynamically_correct,
         }
 
+        return dict(filter(lambda item: item[1] is not None, params.items()))
```

Comment:
any reason to use `filter` instead of a comprehension?
```
{key: value for key, value in params.items() if value is not None}
```

Reply:
no reason, I'll swap it for a comprehension